website/docs/components/code-block/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,100 @@
+## Usage
+
+### When to use
+
+- When displaying code examples and longer snippets of code that benefit from syntax highlighting.
+
+### When not to use
+
+- As a full-blown code editor
+- When displaying keys, tokens, or other generated values, use a [CopySnippet](/components/copy/snippet) instead.
+
+## Inline
+
+The `isInline` property increases the portability of the CodeBlock to ensure that it can be used in different contexts. For example, a common use case of the CodeBlock is in a "standalone" context, which can be part of a form, multi-step process, and is generally a part of the normal layout flow.
```

Comment:
https://github.com/hashicorp/design-system/pull/1691/files#r1367334741

Reply:
Resolved in https://github.com/hashicorp/design-system/pull/1749/commits/19a092530be5f2419739f829a8d000a99a7cb4a7